discovery/kubernetes/kubernetes.go

```diff
@@ -809,13 +809,13 @@ func (d *Discovery) newEndpointsByNodeInformer(plw *cache.ListWatch) cache.Share
 		return nodes, nil
 	}
 
-	return cache.NewSharedIndexInformer(plw, &apiv1.Endpoints{}, resyncDisabled, indexers)
+	return d.mustNewSharedIndexInformer(plw, &apiv1.Endpoints{}, resyncDisabled, indexers)
 }
 
 func (d *Discovery) newEndpointSlicesByNodeInformer(plw *cache.ListWatch, object runtime.Object) cache.SharedIndexInformer {
 	indexers := make(map[string]cache.IndexFunc)
 	if !d.attachMetadata.Node {
-		cache.NewSharedIndexInformer(plw, &disv1.EndpointSlice{}, resyncDisabled, indexers)
```

Comment:
aren't we missing a `return` here?
wasn't caught by any linter? test?

Reply:
Good spot.  

I didn't see this issue described in any of the linters here: https://github.com/mgechev/revive/blob/master/RULES_DESCRIPTIONS.md

Reply:
Yes, I was expecting staticcheck's https://github.com/dominikh/go-tools/blob/a367d57f6820a9e2458d1cc356efe3a979419721/staticcheck/sa4017/sa4017.go#L19 as`cache.NewSharedIndexInformer` seems to be pure, I'll take a look there.

For now, I'll just fix this in another PR and I'll try to cover it by a test.

Reply:
here we go https://github.com/prometheus/prometheus/pull/13593